addons/miscMedical/XEH_postInit.sqf

```diff
@@ -71,11 +71,378 @@ if (hasInterface) then {
+
+    private _alpha = 808 * (1 + (_scaledDistance / 4.5)^2);
+    private _beta = sqrt (1 + (_scaledDistance / 0.048)^2);
+    private _gamma = sqrt (1 + (_scaledDistance / 0.32)^2);
+    private _delta = sqrt (1 + (_scaledDistance / 1.35)^2);
+    
+    private _finalPressure = _ambientPressure * _alpha / (_beta * _gamma * _delta);
+    _finalPressure
+};
 
+FUNC(getChanceOfDeath) = {
+    // percent chances calculated from https://www.cdc.gov/niosh/docket/archive/pdfs/niosh-125/125-explosionsandrefugechambers.pdf
+    params ["_pressure"];
+    private _psi = _pressure / 6895;
+    linearConversion [35, 65, _psi, 0, 1, true];
+};
+
+DFUNC(generateVisuals) = {
+    params ["_position", "_explosiveMass", "_filler"];
+    
+    private _distances = [0.05, 0.2, 0.4, 0.6, 0.8, 1, 1.2, 1.4, 1.6, 1.8, 2, 3, 5, 10];
+
+    private _tntEquivalent = [_explosiveMass, _filler] call FUNC(calculateTNTEquivalent);
 
+    private _blastData = [];
+    {
+        private _scaledDistance = [_x, _tntEquivalent] call FUNC(calculateScaledDistance);
+        private _pressure = [_scaledDistance] call FUNC(calculatePressure);
+        
+        private _chanceOfFatality = [_pressure] call FUNC(getChanceOfDeath);
 
+        _blastData pushBack [_x, _pressure, _chanceOfFatality];
+    } forEach _distances;
+
+    [{
+        _this params ["_args", "_handle"];
+        _args params ["_position", "_blastData", "_stopTime"];
+        
+        if (_stopTime >= 0 && { CBA_missionTime > _stopTime }) exitWith {
+            [_handle] call CBA_fnc_removePerFrameHandler;
+        };
+        
+        private _maxIconsPerRing = 6;
+        {
+            _x params ["_distance", "_pressureAtPosition", "_chanceOfFatality"];
+            private _iconCount = _maxIconsPerRing;
+            if (_distance == 0) then {
+                _iconCount = 1;
+            };
+            
+            _chanceOfFatality = 0 max (_chanceOfFatality min 1);
+            
+            private _colour = vectorLinearConversion [0, 1, _chanceOfFatality, [1, 1, 1], [1, 0, 0]];
+            _colour pushBack 1;
+            
+            for "_i" from 0 to (_maxIconsPerRing - 1) do {
+                private _angle = _i * 2 * pi / _maxIconsPerRing;
+                
+                private _iconPosition = _position getPos [_distance, deg _angle];
+                
+                drawIcon3D [
+                    "\a3\ui_f\data\IGUI\Cfg\Cursors\explosive_ca.paa",
+                    _colour,
+                    _iconPosition,
+                    0.2, 0.2, 1,
+                    format ["Distance: %1m Pressure: %2Pa Fatality Chance: %3 PsI: %4", _distance, _pressureAtPosition, _chanceOfFatality, _pressureAtPosition / 6895]
+                ];
+            };
+        } forEach _blastData;
+    }, 0, [_position, _blastData, CBA_missionTime + 10]] call CBA_fnc_addPerFrameHandler
+};
+
+if (isServer) then {
+    // add explosion event handler
+    [QACEGVAR(frag,frag_eh), {
```

Comment:
I think there might be some thermobaric rounds that we'd want to over-pressure
but are set to not ace_frag, we might be able to tweak their configs to always frag track, but not create any elements

we can deal with this later